graph/src/data/subgraph/features.rs

```diff
@@ -0,0 +1,173 @@
+//! Functions to detect subgraph features.
+//!
+//! The rationale of this module revolves around the concept of feature declaration and detection.
+//!
+//! Features are declared in the `subgraph.yml` file, also known as the subgraph's manifest, and are
+//! validated by a graph-node instance during the deploy phase or by direct request.
+//!
+//! A feature validation error will be triggered in either one of the following cases:
+//! 1. Using undeclared features.
+//! 2. Declaring a nonexistent feature name.
+//!
+//! Feature validation is performed by the [`validate_subgraph_features`] function.
+
+use crate::{
+    blockchain::Blockchain,
+    data::{graphql::DocumentExt, schema::Schema, subgraph::SubgraphManifest},
+    prelude::{Deserialize, Serialize},
+};
+use itertools::Itertools;
+use std::{collections::BTreeSet, fmt, str::FromStr};
+
+const IPFS_ON_ETHEREUM_CONTRACTS_FUNCTION_NAMES: [&'static str; 2] = ["ipfs.cat", "ipfs.map"];
```

Comment:
I think it's useful to have a comment above this with an UUID that should be [here](https://github.com/graphprotocol/graph-node/blob/704fefc870ec2ed93a48467dab601999a0029334/runtime/wasm/src/module/mod.rs#L441) as well. 

Also informing that whenever a new `ipfs` function is added to the `host-exports`, we should add here as well.

What do you think?